### PR TITLE
crowbar: Fix upgrade from 3.0

### PR DESCRIPTION
**Pull request by nicolasbock**
Small fix for https://github.com/crowbar/crowbar-core/pull/486.

### CODE REVIEW

**Comment by nicolasbock** [comment]
@vuntz The cloudupgrade migrates the barclamps now. However, the upgrade still fails for some reason. The error message is `Schema migration failed: Template for openstack does not exist`. I don't understand why `openstack` is tried here at all. I am investigating...

**Comment by nicolasbock** [comment]
For some reason, `BarclampCatalog.barclamps` contains `openstack`, which is what leads to the failure.

File: crowbar_framework/lib/schema_migration.rb [hunk 1]
```diff
@@ -25,7 +25,11 @@ def self.run
   end
 
   def self.run_for_bc bc_name
-    template = Proposal.new(barclamp: bc_name)
+    begin
+      template = Proposal.new(barclamp: bc_name)
+    rescue StandardError => e
```
Comment (nicolasbock) [inline crowbar_framework/lib/schema_migration.rb]: I'll fix it...

File: crowbar_framework/lib/schema_migration.rb [hunk 2]
```diff
@@ -25,7 +25,11 @@ def self.run
   end
 
   def self.run_for_bc bc_name
-    template = Proposal.new(barclamp: bc_name)
+    begin
+      template = Proposal.new(barclamp: bc_name)
+    rescue StandardError
```
Comment (vuntz) [inline crowbar_framework/lib/schema_migration.rb]: Don't catch `StandardError`, but `TemplateMissing` :-)

Comment (nicolasbock) [inline crowbar_framework/lib/schema_migration.rb]: Ah, and I thought I was being smart to avoid `Exception` :smile:

**Comment by vuntz** [comment]
+1

**Comment by aplanas** [comment]
+1

**Comment by nicolasbock** [comment]
I ran through a full upgrade on mkcloud1. It's working.

**Pull request merged**